src/Tribe/REST/V1/Post_Repository.php

```diff
@@ -280,6 +275,43 @@ public function get_organizer_data( $event_or_organizer_id ) {
 		return $single ? reset( $data ) : $data;
 	}
 
+	/**
+	 * Adds the Global ID fields to a set of rest data
+	 *
+	 * @param array  $data  Rest Array of data
+	 * @param int    $id    Post ID
+	 *
+	 * @return array
+	 */
+	protected function add_global_id_fields( $data, $post_id ) {
+		$global_id = new Tribe__Utils__Global_ID;
+		$global_id->type( 'url' );
+		$global_id->origin( home_url() );
+
+		$lineage = get_post_meta( $post_id, Tribe__Events__Aggregator__Event::$global_id_lineage_key );
+
+		$data['global_id'] = $global_id->generate( [ 'id' => $post_id ] );
+		$data['global_id_lineage'] = array_merge( (array) $data['global_id'], (array) $lineage );
+
+		return $data;
+	}
+
+	/**
+	 * @param string $date A date string in a format `strtotime` can parse.
+	 *
+	 * @return array
+	 */
+	protected function get_date_details( $date ) {
+		return array(
+			'year'    => date( 'Y', strtotime( $date ) ),
```

Comment:
* _#fixed_
* Yes, `date()` is the right choice because we need to be able to do both based on whatever was passed as the `$date` param. Check lines [#90](https://github.com/moderntribe/the-events-calendar/blob/5e6f717f745dc0d70e9c5c3c37110f44b2e5c00f/src/Tribe/REST/V1/Post_Repository.php#L90) and [#86](https://github.com/moderntribe/the-events-calendar/blob/5e6f717f745dc0d70e9c5c3c37110f44b2e5c00f/src/Tribe/REST/V1/Post_Repository.php#L86) of that same file.